doctrine/events.rst

```diff
@@ -0,0 +1,345 @@
+         * @ORM\HasLifecycleCallbacks()
+         */
+        class Product
+        {
+            // ...
+
+            /**
+             * @ORM\PrePersist
+             */
+            public function setCreatedAtValue()
+            {
+                $this->createdAt = new \DateTime();
+            }
+        }
+
+    .. code-block:: yaml
+
+        # config/doctrine/Product.orm.yml
+        App\Entity\Product:
+            type: entity
+            # ...
+            lifecycleCallbacks:
+                prePersist: ['setCreatedAtValue']
+
+    .. code-block:: xml
+
+        <!-- config/doctrine/Product.orm.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping
+                https://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
+
+            <entity name="App\Entity\Product">
+                <!-- ... -->
+                <lifecycle-callbacks>
+                    <lifecycle-callback type="prePersist" method="setCreatedAtValue"/>
+                </lifecycle-callbacks>
+            </entity>
+        </doctrine-mapping>
+
+.. note::
+
+    Some lifecycle callbacks receive an argument that provides access to
+    useful information such as the current entity manager (e.g. the ``preUpdate``
+    callback receives a ``PreUpdateEventArgs $event`` argument).
+
+.. _doctrine-listener:
+
+Doctrine Lifecycle Listeners
+----------------------------
+
+Lifecycle listeners are defined as PHP classes that listen to a single Doctrine
+event on all the application entities. For example, suppose that you want to
+update some search index whenever a new entity is persisted in the database. To
+do so, define a listener for the ``postPersist`` Doctrine event::
+
+    // src/EventListener/SearchIndexer.php
+    namespace App\EventListener;
+
+    use App\Entity\Product;
+    // for Doctrine < 2.4: use Doctrine\ORM\Event\LifecycleEventArgs;
+    use Doctrine\Common\Persistence\Event\LifecycleEventArgs;
+
+    class SearchIndexer
+    {
+        public function postPersist(LifecycleEventArgs $args)
+        {
+            $entity = $args->getObject();
+
+            // if this listener only applies to certain entity types,
+            // add some code to check the entity type as early as possible
+            if (!$entity instanceof Product) {
+                return;
+            }
+
+            $entityManager = $args->getObjectManager();
+            // ... do something with the Product entity
+        }
+    }
+
+The listener methods receive a ``LifecycleEventArgs`` object as argument, which
+gives you access to both the entity object of the event and the entity manager
+itself.
+
+.. tip::
+
+    Doctrine 2.4 introduced a new feature called `Entity Listeners`_ which allow
```

Comment:
this should deserve its own section actually, because the way to register them is different.